src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/util/UserSelector.java

```diff
@@ -91,7 +93,9 @@ public String getDisplayName() {
         }
 
         @Restricted(NoExternalUse.class) // Stapler only
+        @RequirePOST
         public FormValidation doCheckSelectedUserId(@QueryParameter String selectedUserId) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Comment:
from a discussion in https://github.com/jenkinsci/jenkins/pull/9598 currently any user with `Jenkins.READ` permission has access to user ids. So it would be sufficient to lower this to `Jenkins.READ`

Reply:
Or apply the same approach as for the GroupSelector with `@AncestorInPath`